js/scripts/zone_loaders/village_zone_loaders.js

```diff
@@ -415,22 +304,30 @@ const loadVillageMain = () => {
         GAME.addEntity(new Mayor(Vector.blockToWorldSpace(blockPosMayor), new Conversation(getAllConversationArrays().playground.papaChad.testNoChoices)), 0);
 
 
-        // Place chad.
+        let weather = "warm";
+        let surfaceSnow = false;
+        if (weather === "snow") {
+            surfaceSnow = true
+        }
+        new tilemapInterpreter(villageMainTileMap, surfaceSnow);
+        new WeatherSystem(weather, 2, "day");
```

Comment:
Why are these constructors? They are being evoked essentially as methods and the constructed instances are never used. Unless you expect to use either of these as objects, these should just be static methods.

Reply:
Gotcha, why is that? I can change that, I don't understand the benefit of changing it. That being said I also don't have an issue with changing it.

Reply:
The main reason is that you're not using the resulting instance, which is the whole point of a constructor. Calling a constructor without using the result is essentially the same thing as using a static method.

Reply:
I am concerned that changing it to using static methods and referencing the class in a static context will cause issues. If you set weather to rain in one zone and the zone is disposed of and the new zone sets the weather to snow then would it not cause rain and snow to both fall because of the static reference? Inside of the weather system entities are added and if we make the function call unnecessarily multiple times add the weather after setting it we would be adding in the rain or snow if I'm not mistaken and it will never remove the rain or snow when it needs to as the function calls are not static references. Is there something missing in my logic? I'll take a look and see if I can make the change you're suggesting however I'll be working all weekend and would suggest that you commit the changes you are suggesting to this PR if I am unable to complete the changes or convince you otherwise in a timely manner.

Reply:
I made the change. You were correct that this works however I don't understand what issue there is with instances in this situation. I'll push my changes soon. Thanks for the feedback!

Reply:
Thanks for making the change. It's certainly not a big deal, and like I said it wasn't causing any issues functionality wise as it was, but if you were using a constructor as if it was a static method (which you were), why not use a static method? The *entire* point of a constructor is to create a new instance. If you don't need an instance, you shouldn't be using a constructor. To me, it's bad practice in general to use a programming construct improperly like that without a good reason, because it can be unclear to others reading your code what you're trying to do, lead to poor design choices, and potentially have unintended side effects caused by the functionality of the specific construct.